op-node/rollup/derive/attributes.go

```diff
@@ -116,14 +116,27 @@ func (ba *FetchingAttributesBuilder) PreparePayloadAttributes(ctx context.Contex
 		upgradeTxs = append(upgradeTxs, fjord...)
 	}
 
-	l1InfoTx, err := L1InfoDepositBytes(ba.rollupCfg, sysConfig, seqNumber, l1Info, nextL2Time)
+	var hasDeposits = len(depositTxs) > 0
+	l1InfoTx, err := L1InfoDepositBytes(ba.rollupCfg, sysConfig, seqNumber, l1Info, nextL2Time, hasDeposits)
 	if err != nil {
 		return nil, NewCriticalError(fmt.Errorf("failed to create l1InfoTx: %w", err))
 	}
 
```

Comment:
we could simplify this by doing something like:

```go
var postDeposits []hexutil.Bytes
if ba.rollupCfg.IsIsthmus(nextL2BlockTime) {
  depositsCompleteTx := DepositsCompleteTx()
  postDeposits = append(postDeposits, depositsCompleteTx)
}

txs := make([]hexutil.Bytes, 0, 1+len(depositTxs)+len(upgradeTxs)+len(postDeposits))
```

Reply:
perfect, way cleaner!